src/components/App.tsx

```diff
@@ -236,15 +245,27 @@ export class App extends React.Component<any, AppState> {
     }
   };
 
-  private initializeSocketClient = () => {
+  private initializeSocketClient = (
+    opts: { showLoadingState: boolean } = {
+      showLoadingState: true,
+    },
+  ) => {
     if (this.socket) {
       return;
     }
     const roomMatch = getCollaborationLinkData(window.location.href);
     if (roomMatch) {
-      this.setState({
-        isCollaborating: true,
-      });
+      let isInitialized = !opts.showLoadingState;
+      const initialize = () => {
+        clearTimeout(initializationTimer);
+        isInitialized = true;
+        if (this.state.isLoading && !this.unmounted) {
+          this.setState({ isLoading: false });
```

Comment:
This causes a React warning:

> Warning: An update (setState, replaceState, or forceUpdate) was scheduled from inside an update function. Update functions should be pure, with zero side-effects. Consider using componentDidUpdate or a callback.

in the case where the init times out, e.g. when none of the clients are broadcasting init `UPDATE_SCENE` message.

Anyone has a clue how to fix it? If not, I think it's not really important.